tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -82,6 +82,67 @@ private void SetupMocks(MockHttpManager httpManager)
         private void SetupMocks(MockHttpManager httpManager, string authority)
         {
             httpManager.AddInstanceDiscoveryMockHandler();
+        }
+
+        [DataTestMethod]
+        [DataRow(true, null, true)]
+        [DataRow(false, null, false)]
+        [DataRow(null, null, false)] // the default is false
+        [DataRow(null, true, true)]
+        [DataRow(null, false, false)]
+        [DataRow(true, true, true)]
+        [DataRow(false, false, false)]
+        [DataRow(true, false, false)] // request overrides
+        [DataRow(false, true, true)] // request overrides
+        public async Task JsonWebTokenWithX509PublicCertSendCertificateTest2Async(bool? appFlag, bool? requestFlag, bool expectX5c)
+        {
+            using (var harness = CreateTestHarness())
+            {
+                SetupMocks(harness.HttpManager);
+                var certificate = new X509Certificate2(
+                ResourceHelper.GetTestResourceRelativePath("valid_cert.pfx"),
+                TestConstants.DefaultPassword);
+
+                var appBuilder = ConfidentialClientApplicationBuilder
+                .Create(TestConstants.ClientId)
+                .WithHttpManager(harness.HttpManager);
+
+                appBuilder = appBuilder.WithCertificate(certificate); // app flag
+                if (appFlag.HasValue)
+                {
+                    appBuilder.Config.SendX5C = appFlag.Value;
+                }
+
+                var app = appBuilder.BuildConcrete();
+
+                //Check for x5c claim
+                if (expectX5c)

```

Comment:
If I understand this, we only validate that `x5c` is added when we expect it to be added. But if `expectX5c` is false, we don't verify that it is not added?

Reply:
Good catch. Done